src/components/calcite-input/calcite-input.tsx

```diff
@@ -183,27 +240,270 @@ export class CalciteInput {
   }
 
   componentWillLoad(): void {
-    this.defaultValue = this.value;
     this.childElType = this.type === "textarea" ? "textarea" : "input";
-    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
-    this.minString = this.min?.toString();
+    this.defaultValue = this.value;
     this.maxString = this.max?.toString();
-    this.stepString = this.step?.toString();
+    this.minString = this.min?.toString();
+    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
   }
 
   componentDidLoad(): void {
     this.slottedActionEl = this.el.querySelector("[slot=input-action]");
     this.setDisabledAction();
+    if (this.shouldFormatNumberByLocale()) {
+      this.childEl.style.cssText = hiddenInputStyle;
+    }
   }
 
-  get isTextarea(): boolean {
-    return this.childElType === "textarea";
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputFocus: EventEmitter;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputBlur: EventEmitter;
+
+  /**
+   * This event fires when the value of the input changes.
+   */
+  @Event({ eventName: "calciteInputInput", cancelable: true }) calciteInputInput: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    if (this.isClearable && getKey(event.key) === "Escape") {
+      this.clearInputValue(event);
+      event.preventDefault();
+    }
   }
 
-  get isClearable(): boolean {
-    return !this.isTextarea && (this.clearable || this.type === "search") && this.value.length > 0;
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** focus the rendered child element */
+  @Method()
+  async setFocus(): Promise<void> {
+    if (this.shouldFormatNumberByLocale()) {
+      this.childNumberEl?.focus();
+    } else {
+      this.childEl?.focus();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clearInputValue = (event: KeyboardEvent | MouseEvent): void => {
+    const calciteInputInputEvent = this.emitInputFromUserInteraction("", event);
```

Comment:
Looks like the emit => defaultPrevented? => setValue logic could also be extracted into a method.

Reply:
Follow up issue?  I'd like to also maybe add some more tests too, but I'd like to get this merged in today.

Reply:
👍🏼